src/lightning/__init__.py

```diff
@@ -28,3 +29,15 @@
     "seed_everything",
     "Fabric",
 ]
+
+
+def _cli_entry_point() -> None:
+    from lightning_utilities.core.imports import RequirementCache
+
+    if not RequirementCache("lightning.app"):
```

Comment:
Shouldn't this use `ModuleAvailableCache` instead? Or https://github.com/Lightning-AI/utilities/blob/main/src/lightning_utilities/core/imports.py#L113 since https://github.com/Lightning-AI/utilities/pull/147 was merged. You might need to check the current utilities version to avoid the deprecation added, or upgrade it

Reply:
Sorry, why would I want to use ModuleAvailableCache if it is deprecated?

Reply:
It depends on the utilities version.  If the latest version is used then you'd need RequirementCache(module=...)

Reply:
The fact that we deprecated ModuleAvailableCache is outrageously annoying to me. It seems very unnecessary and will create warnings for users of older lightning versions which used this piece of code.

Reply:
I totally understand. I also did not agree :/